test/src/main/java/org/apache/accumulo/test/functional/HalfClosedTabletIT.java

```diff
@@ -0,0 +1,279 @@
+
+    try (final var client = Accumulo.newClient().from(getClientProps()).build()) {
+      final var tops = client.tableOperations();
+      tops.create(tableName);
+      TableId tableId = TableId.of(tops.tableIdMap().get(tableName));
+      try (final var bw = client.createBatchWriter(tableName)) {
+        final var m1 = new Mutation("a");
+        final var m2 = new Mutation("b");
+        m1.put(new Text("cf"), new Text(), new Value());
+        m2.put(new Text("cf"), new Text(), new Value());
+        bw.addMutation(m1);
+        bw.addMutation(m2);
+      }
+
+      setInvalidClassLoaderContextPropertyWithoutValidation(getCluster().getServerContext(),
+          tableId);
+
+      // Need to wait for TabletServer to pickup configuration change
+      Thread.sleep(3000);
+
+      Thread configFixer = new Thread(() -> {
+        UtilWaitThread.sleep(3000);
+        removeInvalidClassLoaderContextPropertyWithoutValidation(getCluster().getServerContext(),
+            tableId);
+      });
+
+      long t1 = System.nanoTime();
+      configFixer.start();
+
+      // The split will probably start running w/ bad config that will cause it to get stuck.
+      // However once the config is fixed by the background thread it should continue.
+      tops.addSplits(tableName, Sets.newTreeSet(List.of(new Text("b"))));
+
+      long t2 = System.nanoTime();
+      // expect that split took at least 3 seconds because that is the time it takes to fix the
+      // config
+      assertTrue(TimeUnit.NANOSECONDS.toMillis(t2 - t1) >= 3000);
+
+      // offline the table which will unload the tablets. If the context property is not
+      // removed above, then this test will fail because the tablets will not be able to be
+      // unloaded
+      tops.offline(tableName);
+
+      Wait.waitFor(() -> countHostedTablets(client, tableId) == 0L, 340_000);
+    }
+  }
+
+  @Test
+  public void testIteratorThrowingTransientError() throws Exception {
+
+    // In this scenario a minc iterator throws an error some number of time, then
+    // succeeds. We want to verify that the minc is being retried and the tablet
+    // can be closed.
+
+    try (AccumuloClient c = Accumulo.newClient().from(getClientProps()).build()) {
+
+      String tableName = getUniqueNames(1)[0];
+      final var tops = c.tableOperations();
+
+      tops.create(tableName);
+      final var tid = TableId.of(tops.tableIdMap().get(tableName));
+
+      try (BatchWriter bw = c.createBatchWriter(tableName)) {
+        Mutation m = new Mutation(new Text("r1"));
+        m.put("acf", tableName, "1");
+        bw.addMutation(m);
+      }
+
+      IteratorSetting setting = new IteratorSetting(50, "error", ErrorThrowingIterator.class);
+      setting.addOption(ErrorThrowingIterator.TIMES, "3");
+      c.tableOperations().attachIterator(tableName, setting, EnumSet.of(IteratorScope.minc));
+      c.tableOperations().compact(tableName, new CompactionConfig().setWait(true));
```

Comment:
I know this is the default, feel like it documents intent in the test.

```suggestion
      c.tableOperations().compact(tableName, new CompactionConfig().setWait(true).setFlush(true));
```